client/client.go

```diff
@@ -213,17 +216,19 @@ func (c *authenticatingClient) MakeServiceURL(serviceType, apiVersion string, pa
 	if !ok {
 		return "", errors.New("no endpoints known for service type: " + serviceType)
 	}
-	requestedVersion, err := parseVersion(apiVersion)
-	if err != nil {
-		return "", err
-	}
-	apiURLVersionInfo, err := c.getAPIVersions(serviceURL)
-	if err != nil {
-		return "", err
-	}
-	serviceURL, err = getAPIVersionURL(apiURLVersionInfo, requestedVersion)
-	if err != nil {
-		return "", err
+	if apiVersion != "" {
```

Comment:
maybe invert this to have the indented block
if apiVersion == "" {
  return blah
}
// do version stuff
...